### PR TITLE
Move source icon in incident detail to the header

### LINKED ISSUE #161: Move source icon in incident detail to the header
Objects are tied to single sources by now. There's no reason to show the source icon for every history entry anymore.

But since it's tied to the object, it should instead be shown in the header alongside the object's name. (i.e. Aligned right, as previously)

A beneficial side effect should be, that the icon is also visible in the primary incident list from there on.

### PR DESCRIPTION
**Pull request by raviks789**
fixes #161

### CODE REVIEW

**Review by sukhwinder33445 — CHANGES_REQUESTED** [review]
- The icon is no longer aligned in the center due to this change (https://github.com/Icinga/icinga-notifications-web/commit/bae69e5cbb08add2cf5978e76427356b0875aa5a)
- The icons in the list should be statically aligned.
Screenshot 2024-04-24 at 11 48 58

**Comment by ncosta-ic** [comment]
> * The icon is no longer aligned in the center due to this change ([bae69e5](https://github.com/Icinga/icinga-notifications-web/commit/bae69e5cbb08add2cf5978e76427356b0875aa5a))
> 
> * The icons in the list should be statically aligned.
>

I stumbled accross the same problem and fixed it by adding a `margin-top: .15em` to the `i.icon` rule for `.source-icon` in `public/css/common.less`
I will not include this change in my PR (https://github.com/Icinga/icinga-notifications-web/pull/166) but it should be an easy fix.

**Comment by sukhwinder33445** [comment]
> I stumbled accross the same problem and fixed it by adding a `margin-top: .15em` to the `i.icon` rule for `.source-icon` in `public/css/common.less`

It should also be fixed if you only remove this change. ([bae69e5](https://github.com/Icinga/icinga-notifications-web/commit/bae69e5cbb08add2cf5978e76427356b0875aa5a))

**Comment by ncosta-ic** [comment]
> > I stumbled accross the same problem and fixed it by adding a `margin-top: .15em` to the `i.icon` rule for `.source-icon` in `public/css/common.less`
> 
> It should also be fixed if you only remove this change. ([bae69e5](https://github.com/Icinga/icinga-notifications-web/commit/bae69e5cbb08add2cf5978e76427356b0875aa5a))
source_icon_alignment
(The red box should be aligned with to the cyan guides)

I quickly compared all three variants:
- vertical-align: top
- vertical-align: top, margin-top: .15em
- without both css rules

From my point of view, the second option is also not quite right, but it's much closer to be perfectly aligned than the other two variants.

File: public/css/list/incident-list.less [hunk 1]
```diff
@@ -0,0 +1,6 @@
+.incident-list .list-item {
```
Comment (nilmerg) [inline public/css/list/incident-list.less:1]: This doesn't seem to be specific to the incident list. The event list could also need it, so please move this to common.less

**Review by nilmerg — CHANGES_REQUESTED** [review]
The placement of the icon should be fixed separately.

File: library/Notifications/Widget/ItemList/IncidentHistoryListItem.php [hunk 2]
```diff
@@ -76,11 +76,6 @@ protected function assembleHeader(BaseHtmlElement $header): void
         }
 
         $header->addHtml($this->createCaption());
-        if ($this->item->type === 'opened' || $this->item->type === 'incident_severity_changed') {
-            $header->add(
-                (new SourceIcon(SourceIcon::SIZE_BIG))->addHtml($this->item->event->object->source->getIcon())
```
[the remark this answers is not in the document]
Comment (raviks789) [reply library/Notifications/Widget/ItemList/IncidentHistoryListItem.php:81]: You were right, I have removed the join now.

File: library/Notifications/Widget/ItemList/IncidentListItem.php [hunk 3]
```diff
@@ -73,19 +80,22 @@ protected function assembleTitle(BaseHtmlElement $title): void
...
-                ['class' => 'meta'],
-                [
-                    'closed ',
-                    new TimeAgo($this->item->recovered_at->getTimestamp())
-                ]
-            ));
+            $meta->add([
+                $this->translate('closed') . ' ',
+                new TimeAgo($this->item->recovered_at->getTimestamp())
+            ]);
```
Comment (nilmerg) [inline library/Notifications/Widget/ItemList/IncidentListItem.php:93]: With this, a translator has the option to move the time representation.

```suggestion
           $meta->addHtml(FormattedString::create(
               $this->translate('closed %s', '(incident) ... <relative time>'),
               new TimeAgo($this->item->recovered_at->getTimestamp())
           ));
```

**Review by nilmerg — CHANGES_REQUESTED** [review]
> The placement of the icon should be fixed separately.

With this, I meant the placement of the icon inside the circle. Essentially, my answer to the discussion above.

Now, you've dismissed your own change that added a small gap to the right. I didn't mean you to do this! I still think it's suitable and a visual improvement. *If applied to the incident **and** events list*.

**Review by nilmerg — APPROVED** [review]

**Pull request merged**